src/Microsoft.Dnx.Runtime/Helpers/RoslynAnalyzerHelper.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using Microsoft.Dnx.Compilation;
+using Microsoft.Dnx.Compilation.CSharp;
+using NuGet;
+
+namespace Microsoft.Dnx.Runtime.Helpers
+{
+    internal static class RoslynAnalyzerHelper
+    {
+        private static readonly string AnalyzerBaseFolder = "analyzers";
+        private static readonly string AnalyzerLanguage = "cs";
+
+        public static IList<IAnalyzerReference> GetAnalyerReferences(PackageInfo packageInfo, string packagePath, FrameworkName runtimeFramework)
+        {
+            var files = new List<string>();
+
+            // assumption: analyzers' paths are in following format
+            // analyzer/{framework}/{cs|vb}/{filename}.dll
+            foreach (var relativePath in packageInfo.LockFileLibrary.Files)
+            {
+                if (!relativePath.StartsWith(AnalyzerBaseFolder, StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                if (!relativePath.EndsWith(".dll", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                var parts = relativePath.Split(new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar });
+                if (parts.Length != 4)
+                {
+                    continue;
+                }
+
+                if (!string.Equals(parts[2], AnalyzerLanguage, StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                var frameworkName = VersionUtility.ParseFrameworkFolderName(parts[1]);
+                if (VersionUtility.GetNearest(runtimeFramework, new[] { frameworkName }) != frameworkName)
```

Comment:
I don't think this is right. Instead, you should be able to wrap each file starting `analyzers/` in a class implementing `IFrameworkTargettable` and then pass the whole set of files into GetNearest. It will return to you all the files from one folder, having selected the most appropriate folder. Then you can filter the resulting files by language.

As far as I can tell, this will return analyzers from multiple different folders as long as they are compatible with the runtime folder, which is bad. If there is an `analyzers/net451` and an `analyzers/dotnet` folder, compiling on `dnx451` should **only** pick the analyzers from `analyzers/net451` and **not** any of the analyzers from `analyzers/dotnet`. `GetNearest` should handle that for you.

One other way you could do this is to get the set of folder names directly under `analyzers/` and pass **all** of them in to `GetNearest`. The one returned is the best match and you can scoop up all the analyzers from that folder only.